relay-dynamic-config/src/error_boundary.rs

```diff
@@ -42,6 +42,14 @@ impl<T> ErrorBoundary<T> {
         }
     }
 
+    /// Converts from `ErrorBoundary<T>` to `ErrorBoundary<&T>`.
+    pub fn as_ref(&self) -> ErrorBoundary<&T> {
+        match self {
+            Self::Ok(t) => ErrorBoundary::Ok(t),
+            Self::Err(e) => ErrorBoundary::Err(Arc::clone(e)),
+        }
+    }
+
```

Comment:
It catches and isoltes from deserialization failures in subtrees when deserializing. 

```
struct Foo {
   a: u32,
   b: ErrorBoundary<i32>,
}
```

This will not fail deserialization on a JSON like `{a: 1, b: "this is not a i32"}`, it will deserialize as `Foo { a: 1, b: Error(..) }`.

This is mainly used in project and global configs, small mistakes don't cause the entire system to fail just the subsystem.